src/main/webapp/app/components/sidebar/OncoKBSidebar.tsx

```diff
@@ -15,27 +18,144 @@ const OncoKBSidebar = ({
   showOncoKBSidebar,
   toggleOncoKBSidebar,
   oncoKBSidebarWidth,
+  setOncoKBSidebarWidth,
+  closeNavigationSidebar,
   defaultOpen = false,
 }: IOncoKBSidebarProps) => {
+  const maxWidth = document.body.clientWidth * 0.5;
+  const closeNavWidth = document.body.clientWidth * 0.45;
+
+  const sidebarRef = useRef<HTMLDivElement>(null);
+  const draggableRef = useRef<HTMLDivElement>(null);
+  const sidebarHighlightTimer = useRef(null);
+
+  const [sidebarBorderHovered, setSidebarBorderHovered] = useState(false);
+  const [holdingSidebarBorder, setHoldingSidebarBorder] = useState(false);
+
   useEffect(() => {
     if (defaultOpen) {
       toggleOncoKBSidebar(true);
     }
   }, []);
+
+  useEffect(() => {
+    return () => {
+      setOncoKBSidebarWidth(ONCOKB_SIDEBAR_MIN_WIDTH);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (sidebarRef.current && draggableRef.current) {
+      if (sidebarBorderHovered) {
+        document.body.style.cursor = 'col-resize';
+        highlightSidebarBorder();
+      }
+
+      if (holdingSidebarBorder || sidebarBorderHovered) {
+        highlightSidebarBorder();
+      } else {
+        unhighlightSidebarBorder();
+        document.body.style.cursor = '';
+      }
+    }
+  }, [sidebarBorderHovered, holdingSidebarBorder]);
+
+  function handleMouseDown(mouseDownEvent: React.MouseEvent<HTMLDivElement, MouseEvent>) {
+    mouseDownEvent.preventDefault();
+
+    setHoldingSidebarBorder(true);
+
+    const startX = mouseDownEvent.pageX;
+
+    function onMouseMove(mouseMoveEvent: MouseEvent) {
+      const newWidth = oncoKBSidebarWidth + startX - mouseMoveEvent.pageX;
+
+      if (newWidth > oncoKBSidebarWidth && newWidth > closeNavWidth) {
+        closeNavigationSidebar();
+      }
+
+      if (newWidth <= ONCOKB_SIDEBAR_MIN_WIDTH) {
+        setOncoKBSidebarWidth(ONCOKB_SIDEBAR_MIN_WIDTH);
+      } else if (newWidth >= maxWidth) {
+        setOncoKBSidebarWidth(maxWidth);
+      } else {
+        setOncoKBSidebarWidth(oncoKBSidebarWidth + startX - mouseMoveEvent.pageX);
+      }
+    }
+
+    function onMouseUp() {
+      setHoldingSidebarBorder(false);
+      document.body.removeEventListener('mousemove', onMouseMove);
+    }
+
+    document.body.addEventListener('mousemove', onMouseMove);
+    document.body.addEventListener('mouseup', onMouseUp, { once: true });
```

Comment:
Looks like the `mouseup` listener is not removed? Maybe, the inner functions can be moved one level out and then the useEffect can remove listener when unmounted?